plugins/kubernetes/app/models/kubernetes/deploy_yaml.rb

```diff
@@ -66,9 +66,11 @@ def set_deployment_metadata
       end
     end
 
+    # labels that are used to match the previous deployment,
+    # they cannot change or we will create a new deployment instead of updating the previous one
+    # re-creating roles or re-creating the deploy group will lead to duplicate deployments
     def deployment_labels
-      # Deployment labels should not include the ids of the release, role or deploy groups
-      release_doc_metadata.except(:release_id, :role_id, :deploy_group_id)
+      release_doc_metadata.slice(:project_id, :role_id, :deploy_group_id)
```

Comment:
Good point, I'd lean on the side of validating ... I want to stay out of label management business :)